pkg/restore/crd.go

```diff
@@ -0,0 +1,209 @@
+limitations under the License.
+*/
+
+package restore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rook/kubectl-rook-ceph/pkg/debug"
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func RestoreCrd(ctx context.Context, k8sclientset *k8sutil.Clientsets, operatorNamespace, clusterNamespace string, args []string) {
+	crd := args[0]
+	logging.Info("Detecting which ceph crd resources to restore")
+	getCrName := `kubectl -n %s get %s -o jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.metadata.deletionGracePeriodSeconds}{"\n"}{end}' | awk '$2=="0" {print $1}'`
+	command := fmt.Sprintf(getCrName, clusterNamespace, crd)
+	crName := strings.TrimSpace(exec.ExecuteBashCommand(command))
+
+	if crName == "" {
+		logging.Info("Nothing to do here, no crd resources with deletionGracePeriodSeconds 0")
+		return
+	}
+
+	if len(args) == 2 {
+		crName = args[1]
+	}
+
+	logging.Info("Restoring crd %s name %s", crd, crName)
+
+	logging.Info("Scaling down the operator to 0")
+	err := debug.SetDeploymentScale(ctx, k8sclientset.Kube, operatorNamespace, "rook-ceph-operator", 0)
+	if err != nil {
+		logging.Fatal(err)
+	}
+
+	logging.Info("Backing up kubernetes and crd resources")
+	getCrYamlContent := `kubectl -n %s get %s %s -oyaml > %s.yaml`
+	command = fmt.Sprintf(getCrYamlContent, clusterNamespace, crd, crName, crd)
+	exec.ExecuteBashCommand(command)
+	logging.Info("Backed up crd %s named %s in file %s.yaml", crd, crName, crd)
+
+	webhookConfigName := "rook-ceph-webhook"
+	logging.Info("Deleting validating webhook %s if present", webhookConfigName)
+	err = k8sclientset.Kube.AdmissionregistrationV1().ValidatingWebhookConfigurations().Delete(ctx, webhookConfigName, v1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		logging.Fatal(fmt.Errorf("failed to delete validating webhook %s. %v", webhookConfigName, errors.Wrapf(err, "Operator pod still being scaled down")))
+	}
+
+	logging.Info("Fetching the UID for %s/%s", crd, crName)
+	getCrUID := `kubectl -n %s get %s %s -o 'jsonpath={.metadata.uid}'`
+	command = fmt.Sprintf(getCrUID, clusterNamespace, crd, crName)
+	uid := exec.ExecuteBashCommand(command)
+	logging.Info("Successfully fetched uid %s from %s %s", uid, crd, crName)
+
+	removeOwnerRefOfUID(ctx, k8sclientset, operatorNamespace, clusterNamespace, uid)
+
+	logging.Info("Removing finalizers from %s %s", crd, crName)
+	removeFinalizers := `kubectl -n %s patch %s/%s --type json --patch='[ { "op": "remove", "path": "/metadata/finalizers" } ]'`
+	command = fmt.Sprintf(removeFinalizers, clusterNamespace, crd, crName)
+	logging.Info(exec.ExecuteBashCommand(command))
+
+	logging.Info("Re-creating the crd %s from file %s.yaml created above", crd, crName)
+	recreateCr := `kubectl create -f %s`
```

Comment:
Let's use CR instead of Cr in var names
```suggestion
	recreateCR := `kubectl create -f %s`
```